File: server/core/src/main/java/dev/slimevr/reset/ResetHandler.java
package dev.slimevr.reset;

import java.util.List;
import java.util.concurrent.CopyOnWriteArrayList;


public class ResetHandler {

	private final List<ResetListener> listeners = new CopyOnWriteArrayList<>();

	public ResetHandler() {

	}

	public void sendStarted(int resetType) {
		this.listeners.forEach((listener) -> listener.onStarted(resetType));
	}

	public void sendFinished(int resetType) {
		this.listeners.forEach((listener) -> listener.onFinished(resetType));
	}

	public void addListener(ResetListener listener) {
		this.listeners.add(listener);
	}

	public void removeListener(ResetListener l) {
		listeners.removeIf(listener -> l == listener);
	}
}

File: server/core/src/main/java/dev/slimevr/reset/ResetHandler.kt
package dev.slimevr.reset

import dev.slimevr.protocol.rpc.TransactionInfo
import java.util.concurrent.CopyOnWriteArrayList
import java.util.function.Consumer

class ResetHandler {
	private val listeners: MutableList<ResetListener> = CopyOnWriteArrayList()

	fun sendStarted(resetType: Int, tx: TransactionInfo? = null, bodyParts: List<Int>? = null, progress: Int = 0, duration: Int = 0) {
		this.listeners.forEach { listener: ResetListener -> listener.onStarted(resetType, tx, bodyParts, progress, duration) }
	}

	fun sendFinished(resetType: Int, tx: TransactionInfo? = null, bodyParts: List<Int>? = null, duration: Int) {
		this.listeners.forEach { listener: ResetListener -> listener.onFinished(resetType, tx, bodyParts, duration) }
	}

	fun addListener(listener: ResetListener) {
		this.listeners.add(listener)
	}

	fun removeListener(l: ResetListener) {
		listeners.removeIf { listener: ResetListener -> l === listener }
	}
}
